### PR TITLE
Add something-something-v2 dataset support for video action recognition

### PR DESCRIPTION
**Pull request by bryanyzhu**
**Description:**

Add [something-something-v2](https://20bn.com/datasets/something-something) dataset support for video action recognition. Reproduce performance on TSN and I3D models. 

Model zoo is updated. 

Model store is updated.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-983-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-983/1/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/data/somethingsomethingv2/classification.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    video_loader : bool, default False
+        Whether to use video loader to load data.
+    use_decord : bool, default True
+        Whether to use Decord video loader to load data. Otherwise use mmcv video loader.
+    transform : function, default None
+        A function that takes data and label and transforms them.
+    """
+    def __init__(self,
+                 setting=os.path.expanduser('~/.mxnet/datasets/somethingsomethingv2/train_videofolder.txt'),
+                 root=os.path.expanduser('~/.mxnet/datasets/somethingsomethingv2/20bn-something-something-v2-frames'),
```
Comment (hetong007) [inline gluoncv/data/somethingsomethingv2/classification.py]: seems `setting` is in par with `root`, and `root` is different from doc?

File: scripts/action-recognition/train_recognizer.py [hunk 2]
```diff
@@ -180,6 +180,15 @@ def batch_fn(batch, ctx):
                                                    new_width=opt.new_width, new_height=opt.new_height, new_length=opt.new_length,
                                                    target_width=input_size, target_height=input_size,
                                                    num_segments=opt.num_segments, transform=transform_test)
+    elif opt.dataset == 'somethingsomethingv2':
+        train_dataset = somethingsomethingv2.classification.SomethingSomethingV2(setting=opt.train_list, root=data_dir, train=True,
```
Comment (hetong007) [inline scripts/action-recognition/train_recognizer.py]: line width and indent seems weird, can we import `SomethingSomethingV2` directly?

File: gluoncv/data/somethingsomethingv2/classification.py (hunk 1 above)
Comment (bryanyzhu) [reply gluoncv/data/somethingsomethingv2/classification.py]: Thank you for catching it. Already make `root` the same both in doc and code. 

For the difference between `setting` and `root`, 
`root`: is where the data actually stored
`setting`: is a `.txt` file containing the data path, and used by dataloader.

File: scripts/action-recognition/train_recognizer.py (hunk 2 above)
Comment (bryanyzhu) [reply scripts/action-recognition/train_recognizer.py]: Sure, I will change them all.

**Comment by mli** [comment]
Job PR-983-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-983/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-983-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-983/3/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-983-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-983/4/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**